environs/manual/generate.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2016 Canonical Ltd.
+// Copyright 2016 Cloudbase Solutions SRL
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package manual
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/utils/os"
+	"github.com/juju/utils/series"
+
+	"github.com/juju/juju/cloudconfig/instancecfg"
+	"github.com/juju/juju/environs/manual/common"
+	"github.com/juju/juju/environs/manual/linux"
+)
+
+// NewScriptProvisioner returns a valid ScriptProvisioner that will be used to produce the script.
+func NewScriptProvisioner(icfg *instancecfg.InstanceConfig) (common.ScriptProvisioner, error) {
```

Comment:
I don't see the point of the ScriptProvisioner interface. Why not just have:

```
func ProvisioningScript(icfg *instancecfg.InstanceConfig) (string, error) {
    ...
    switch seriesOS {
    case os.Ubuntu, os.CentOS, os.GenericLinux:
        return linux.ProvisioningScript(icfg), nil
    ...
    }
}
```
?